code/species_defs/species_defs.cpp

```diff
@@ -328,6 +328,31 @@ void parse_species_tbl(const char *filename)
 			if (optional_string("$Countermeasure type:"))
 				stuff_string(species->cmeasure_name, F_NAME, NAME_LENGTH);
 
+			if (optional_string("$Custom wingmate formations:")) {
+
+				while (check_for_string("(")) {
+					species->formations.emplace_back();
+					if (stuff_vec3d_list(species->formations.back().data(), MAX_SHIPS_PER_WING - 1) != MAX_SHIPS_PER_WING - 1) {
+						Warning(LOCATION, "A custom formation for species %s did not have 5 positions. Ignoring.", species->species_name);
+						species->formations.pop_back();
+					}
```

Comment:
I wouldn't write directly to `.data()` here.  Instead, make a temporary `SCP_vector<vec3d> vec_list`, stuff that, then push_back the temp list into the formation list or lists.  See how `parse_waypoint_list` does it.

Reply:
Are there any particular advantages to doing it that way?

Reply:
It's more of a stylistic or code-safety practice.  The `.data()` method returns a pointer to the raw underlying data of the vector, and one should avoid working with the internal structures of other classes in most cases.

My first reaction was that the compiler shouldn't even have allowed this, but upon further examination in *this particular case* it looks safe.  However it's one of those "code smells".

Reply:
Fair enough.